src/formatDate.js

```diff
@@ -1,5 +1,37 @@
 'use strict';
 
+/**
+ * @param {Object} datePartsMatched
+ * @param {string[]} toFormat
+ * @param {string[]} fromFormat
+ *
+ * @returns {Object}
+ */
+
+function convertYear(datePartsMatched, fromFormat, toFormat) {
+  const FOUR_DIGIT_YEAR_FORMAT = 'YYYY';
+  const TWO_DIGIT_YEAR_FORMAT = 'YY';
+
+  if (
+    fromFormat.includes(FOUR_DIGIT_YEAR_FORMAT) &&
+    toFormat.includes(TWO_DIGIT_YEAR_FORMAT)
+  ) {
+    const year = datePartsMatched[FOUR_DIGIT_YEAR_FORMAT];
+
+    datePartsMatched[TWO_DIGIT_YEAR_FORMAT] = year.slice(2);
+  } else if (
+    fromFormat.includes(TWO_DIGIT_YEAR_FORMAT) &&
+    toFormat.includes(FOUR_DIGIT_YEAR_FORMAT)
+  ) {
+    const year = datePartsMatched[TWO_DIGIT_YEAR_FORMAT];
+    const newYear = +year < 30 ? `20${year}` : `19${year}`;
+
+    datePartsMatched[FOUR_DIGIT_YEAR_FORMAT] = newYear;
+  }
+
+  return datePartsMatched;
```

Comment:
The `convertYear` function assumes the year format key exists in the `datePartsMatched` object. However, it relies on the keys being 'YYYY' or 'YY', which might not be the case if the `fromFormat` array has a different order. It's safer to find the index of the year format in the `fromFormat` array and use it to access the corresponding year value in `datePartsMatched`.